src/components/CollectionPagesSection.jsx

```diff
@@ -0,0 +1,111 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import _ from 'lodash';
+
+// Import components
+import OverviewCard from '../components/OverviewCard';
+import ComponentSettingsModal from '../components/ComponentSettingsModal'
+
+// Import styles
+import elementStyles from '../styles/isomer-cms/Elements.module.scss';
+import contentStyles from '../styles/isomer-cms/pages/Content.module.scss';
+
+// Constants
+const RADIX_PARSE_INT = 10;
+
+const CollectionPagesSection = ({ collectionName, pages, siteName }) => {
+    const [isComponentSettingsActive, setIsComponentSettingsActive] = useState(false)
+    const [selectedFile, setSelectedFile] = useState('')
+    const [createNewPage, setCreateNewPage] = useState(false)
+
+    const settingsToggle = (event) => {
+        const { id } = event.target;
+        const idArray = id.split('-');
+
+        // Create new page
+        if (idArray[1] === 'NEW') {
+            setIsComponentSettingsActive((prevState) => !prevState)
+            setSelectedFile('')
+            setCreateNewPage(true)
+        } else {
+          // Modify existing page frontmatter
+          const pageIndex = parseInt(idArray[1], RADIX_PARSE_INT);
+
+          setIsComponentSettingsActive((prevState) => !prevState)
+          setSelectedFile(() => {
+              return isComponentSettingsActive ? null : pages[pageIndex]
+          })
+          setCreateNewPage(false)
+        }
+    }
+
+    return (
+        <>
+            {
+                isComponentSettingsActive
+                && (
+                    <ComponentSettingsModal
+                        modalTitle={"Page Settings"}
+                        settingsToggle={settingsToggle}
+                        category={collectionName}
+                        siteName={siteName}
+                        fileName={selectedFile ? selectedFile.fileName : ''}
+                        isNewFile={createNewPage}
+                        type="page"
+                        pageFilenames={
+                            _.chain(pages)
+                                .map((page) => page.fileName)
+                                .value()
+                        }
+                    />
+                )
+            }
+            <div className={contentStyles.contentContainerBoxes}>
+                {/* Display loader if pages have not been retrieved from API call */}
+                { pages
```

Comment:
The check should be `pages && pages.length > 0` as the pages state is initialised to an empty array

Reply:
Let me check with the designers on whether we should render the "Add new page" button first before the other `OverviewCard`s are rendered, or if we should render them together at the same time

Reply:
Actually there's no need to ask them on something so minor, I made the fix in [1f1b783](https://github.com/isomerpages/isomercms-frontend/pull/206/commits/1f1b7833c729b1123cac0718b475311efb700f43)